### DISCUSSION TITLE
Which image should I bitbake?

### DISCUSSION THREAD

**Discussion by fei4xu** [OE4T/meta-tegra#642] [Q&A] [answered]
Hi,

After `source poky-dunfell/oe-init-build-env build` I got this: 

>You can now run 'bitbake <target>'
Common targets are:
    core-image-minimal
    core-image-sato
    meta-toolchain
    meta-ide-support

and I found another image: `tegra-minimal-initramfs` which inherit `core-image`.

I'd like to check which image should I bitbake to reproduce an image which is close to Nviida's pre-build flash image?

Thanks.

**Comment by madisongh** [comment, marked as answer]
None of the images you build using OE/Yocto Project tools will be very close to the Ubuntu 18.04 image that NVIDIA provides with L4T/JetPack, as the underlying design principles are fundamentally different.  Here, you start small and build in exactly what you need, rather than having a fully-featured set of binaries for a wide variety of use cases that run generically across broad swathes of hardware.

That said, you might want to consider using the [demo distro](https://github.com/OE4T/tegra-demo-distro) we have here, rather than poky,  as a starting point, which has [several images](https://github.com/OE4T/tegra-demo-distro#images) that combine core features with L4T/JetPack contents.

**Reply by fei4xu** [reply]
Hi, thanks for the mentioning of the demo distro.

When I said "a image close to NVidia's", I'm not referring to ubuntu or the GUI part, I was referring to things included in JetPack.

For example **JetPack 4.4**, it has this packages with specified versions:

- TensorRT | 7.1.3  
- cuDNN | 8
- CUDA | 10.2
- Multimedia   API |  
- Computer   Vision | Visionworks 1.6
- OpenCV 4.1.1
- VPI 0.3
- DeepStream 5.0 |  
- GCC 7.3.1

I'd like to check if we can build a image include the same key packages with same version numbers as to certain JetPack version?

**Reply by madisongh** [reply]
[This wiki page](https://github.com/OE4T/meta-tegra/wiki/Which-branch-should-I-use%3F) lists the branches and the L4T/JetPack versions they support. The `dunfell-l4t-r32.4.3` branch would be a good choice for JetPack 4.4 content. There's a corresponding branch in the demo distro I mentioned.

The only thing missing from our JetPack 4.4 support is a a recipe for VPI.

**Reply by sgserg** [reply]
> The only thing missing from our JetPack 4.4 support is a a recipe for VPI.

Is there any place documenting which JetPack components are [not] supported by meta-tegra?

**Reply by ichergui** [reply]
Hey @sgserg 

Please check our [wiki page](https://github.com/OE4T/meta-tegra/wiki/Which-branch-should-I-use%3F)

By using this wiki page, you can align with [L4T](https://developer.nvidia.com/embedded/linux-tegra) to check Jetpack components

**Reply by sgserg** [reply]
Thanks @ichergui, I've seen that page, it lists the versions, but it is only in this issue I see that VPI is missing.
Maybe other versions miss something else?

**Reply by ichergui** [reply]
@sgserg 

First release version of VPI is `1.0.12` that Nvidia provides in `Jetpack 4.5` / `L4T-R32.5.0` 
Do you need it for `Jetpack 4.4` ?   if so, I will discuss with @madisongh and I will add it to `dunfell-l4t-r32.4.3`

**Reply by sgserg** [reply]
@ichergui, turns out we don't use VPI, so no need to add it anywhere on my account, thanks! I was just worried there might be other JetPack components not supported by meta-tegra.

**Comment by notthetup** [comment]
Semi-related question: 

I was trying to build various images `core-image-sato-dev`, `core-image-minimal-dev`, `core-image-full-cmdline` but I only managed to connect to the Jetson over SSH on `core-image-sato-dev`.

Is there a list that explains what each of these images contains? Based on the yocto documentation it's really hard to guess/dechiper why certain images are inaccessible over SSH.

**Reply by lfdmn** [reply]
Try this out
```
bitbake -g <image name> && cat pn-buildlist | grep -ve "native" | sort | uniq
```

**Reply by madisongh** [reply]
Also, if you've already built an image, you'll fine a `.manifest` file for it in the same directory (under `$BUILDDIR/tmp/deploy/images`) where the image is placed.

You can also look at the image recipes themselves, as a starting point.
